File: InternetScanner/languages/php/InternetScanner.php
<?php
namespace BizX\InternetScanner;
final class InternetScanner { public static function classify(string $ip):string { if(filter_var($ip,FILTER_VALIDATE_IP,FILTER_FLAG_IPV4)===false)return 'public'; if(filter_var($ip,FILTER_VALIDATE_IP,FILTER_FLAG_NO_PRIV_RANGE|FILTER_FLAG_NO_RES_RANGE)!==false)return 'public'; return 'local/intranet'; } public static function authorized(string $ip,array $allow):bool{return self::classify($ip)!=='public'||in_array($ip,$allow,true);} }

File: InternetScanner/languages/ruby/internet_scanner.rb
require 'ipaddr'
module BizX
 module InternetScanner
  PRIVATE=[IPAddr.new('10.0.0.0/8'),IPAddr.new('172.16.0.0/12'),IPAddr.new('192.168.0.0/16'),IPAddr.new('127.0.0.0/8'),IPAddr.new('169.254.0.0/16')]
  def self.classify(ip); a=IPAddr.new(ip); PRIVATE.any?{|n|n.include?(a)} ? 'local/intranet' : 'public'; rescue IPAddr::InvalidAddressError; 'public'; end
  def self.authorized(ip,allow); classify(ip)!='public'||allow.include?(ip); end
 end
end
